### PR TITLE
feat: default user tasks with `zeebe:UserTask` extension element

### PR DESCRIPTION
**Pull request by Skaiir**
Related to https://github.com/camunda/camunda-modeler/issues/4648

### CODE REVIEW

**Comment by philippfromme** [comment]
What is the state of this pull request?

**Comment by Skaiir** [comment]
This is currently in draft because there are some failing tests around `userTaskForm`. Getting some help from @philippfromme to get this unblocked.

**Comment by barmac** [comment]
I will look into this now.

**Comment by barmac** [comment]
Since @philippfromme's test update is merged now, let's run it on CI again.

**Comment by barmac** [comment]
Regarding the failing `CleanUpTaskListenersBehaviorSpec` test case, we could remove it as it cannot be reproduced in the UI (task listener on job worker user task).

**Comment by barmac** [comment]
Done via https://github.com/camunda/camunda-bpmn-js-behaviors/pull/86/commits/44dd8bd5e5a78a4f185d731f55ef35f7fb5d80a5

**Comment by barmac** [comment]
I suspect that the remaining test cases might be failing because we change from job worker to zeebe user task, so specifically related to the content of the PR.

What about if we consider changing the editor's behaviour via a popup menu / context pad extension? So that we don't need to add the extension after the element is created, but rather create a proper user task from the beginning? This could have a down (or up?) side of being able to copy and paste a job-worker user task. What do you think @Skaiir ?

**Comment by barmac** [comment]
Whatever solution we take, let's make sure these requirements are met:

* [ ] When I create a user task from scratch, its implementation is set to Zeebe user task
* [ ] When I replace a task with user task, its implementation is set to Zeebe user task
* [ ] When I copy a job worker user task, the implementation type is kept

**Comment by Skaiir** [comment]
I have adjusted the behavior to act on replace, and not do so on copy. Struggled with that second part but GPT found the hint in the context regarding how to differentiate a regular creation from a copy creation.

Then I looked it up in the bpmn-js codebase and it seems to be the proper way: https://github.com/bpmn-io/bpmn-js/commit/bf180321a3a40428c3f87b639b87cc3fc578066e#diff-2f0de25761fb7459e88071f83fd845c5R22.

===

Regarding the leftover failing tests. I don't know this form behavior and I'm not sure what the expectations are, but it seems that it is only applicable for non-zeebe userTasks. The simple fix I could find to clear the last couple tests to green was to bypass the `zeebe:UserTask` behavior in those tests by setting the "copy" hint directly. It basically emulates what would happen in a copy scenario.

**Comment by barmac** [comment]
I'll have a look into this tomorrow.

File: test/camunda-cloud/CreateZeebeUserTaskBehaviorSpec.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+      // then
+      const pastedUserTask = find(elements, (element) =>
+        is(element, 'bpmn:UserTask')
+      );
+
+      const businessObject = getBusinessObject(pastedUserTask),
+            extensionElements = businessObject.get('extensionElements'),
+            zeebeUserTaskExtensions = getZeebeUserTaskElements(pastedUserTask);
+
+      expect(zeebeUserTaskExtensions).to.exist;
```
Comment (Skaiir) [inline test/camunda-cloud/CreateZeebeUserTaskBehaviorSpec.js:135]: This is a little confusing

File: test/camunda-cloud/CreateZeebeUserTaskBehaviorSpec.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        rootElement
+      );
+      bpmnReplace.replaceElement(task, { type: 'bpmn:UserTask' });
+
+      // then
+      const updatedTask = elementRegistry.get(task.id),
+            zeebeUserTaskExtension = getZeebeUserTaskElement(updatedTask);
+
+      expect(zeebeUserTaskExtension).to.exist;
+    }));
```
Comment (Skaiir) [inline test/camunda-cloud/CreateZeebeUserTaskBehaviorSpec.js:167]: Add a negative case for this one

File: test/camunda-cloud/FormsBehaviorSpec.js [hunk 3]
```diff
@@ -191,7 +191,7 @@ describe('camunda-cloud/features/modeling - FormsBehavior', function() {
         });
 
         // when
-        modeling.createShape(element, { x: 100, y: 100 }, rootElement);
+        modeling.createShape(element, { x: 100, y: 100 }, rootElement, { createElementsBehavior: false });
```
Comment (Skaiir) [inline test/camunda-cloud/FormsBehaviorSpec.js:194]: Extract to separate function

**Comment by barmac** [comment]
I've applied the changes ideated in the review session, cf. https://github.com/camunda/camunda-bpmn-js-behaviors/pull/86/commits/dc7860161a444dd1708a91170b57b91a1baea8a7

**Comment by barmac** [comment]
Thanks @Skaiir for providing this

**Comment by barmac** [comment]
Released as [v1.8.0](https://github.com/camunda/camunda-bpmn-js-behaviors/releases/tag/v1.8.0)

**Pull request merged**